File: javarecipe/src/main/java/work/shion/javarecipe/pages/websample/contracts/MainViewContract.java
package work.shion.javarecipe.pages.websample.contracts;

import android.net.Uri;

import work.shion.baser.android.WebViewBuilder;


/**
 * メインコンテンツの表示定義
 */
public interface MainViewContract {

    /**
     * 外部ブラウザに遷移
     */
    void goBrowser(Uri uri);

    /**
     * WebView のセットアップ
     */
    void setupWebView(WebViewBuilder builder);

    /**
     * アラートダイアログの表示
     */
    void showAlert(String message);
}

File: ktrecipe/src/main/java/work/shion/ktrecipe/pages/websample/contracts/MainViewContract.kt
package work.shion.ktrecipe.pages.websample.contracts

import android.net.Uri
import work.shion.baser.android.WebViewBuilder


/**
 * メインコンテンツの表示定義
 */
internal interface MainViewContract {

    /**
     * 外部ブラウザに遷移
     */
    fun goBrowser(uri: Uri)

    /**
     * WebView のセットアップ
     */
    fun setupWebView(builder: WebViewBuilder)

    /**
     * アラートダイアログの表示
     */
    fun showAlert(message: String)
}
